include/circt/Dialect/FIRRTL/FIRRTLDeclarations.td

```diff
@@ -129,6 +129,21 @@ def InstanceOp : HardwareDeclOp<"instance", [
     /// the new InstanceOp to the same location.
     InstanceOp cloneAndInsertPorts(ArrayRef<std::pair<unsigned, PortInfo>> ports);
 
+    //===------------------------------------------------------------------===//
+    // Instance graph methods
+    //===------------------------------------------------------------------===//
+
+    // Quick lookup of the referenced module using the instance graph.
+    template <typename T = ::circt::igraph::ModuleOpInterface>
+    T getReferencedModuleByGraph(::circt::igraph::InstanceGraph &instanceGraph) {
+      auto moduleNameAttr = getModuleNameAttr().getAttr();
+      auto *node =  instanceGraph.lookup(moduleNameAttr);
+      if (!node)
+        return nullptr;
+      Operation *moduleOp = node->getModule();
+      return dyn_cast_or_null<T>(moduleOp);
```

Comment:
Can this ever fail for valid IR?  dyn_cast<T> seems better.

Reply:
Same with !node.

Reply:
Yes, in the `hw` version there is a virtual null node that points to the public modules.